pmd-core/src/main/java/net/sourceforge/pmd/RuleSet.java

```diff
@@ -169,6 +169,7 @@ public RuleSetBuilder addRuleReplaceIfExists(final Rule rule) {
             for (final Iterator<Rule> it = rules.iterator(); it.hasNext();) {
                 final Rule r = it.next();
                 if (r.getName().equals(rule.getName()) && r.getLanguage() == rule.getLanguage()) {
+                    LOG.warning("The rule with name " + rule.getName() + " was duplicated and has been overwritten.");
```

Comment:
I don't think this should be handled in the RulesetBuilder, but rather in the RulesetFactory.

For example, one case where this might be noisy is the following:
```xml
<rule ref="myruleset.xml" />
<rule ref="myruleset.xml/SomeRule" >
  <!-- some config -->
</rule>
```
It's unnecessary to explicitly exclude `SomeRule` from the first rule element and has been forever.
The info should be somehow preserved in the RulesetFactory, but would be anyway unavailable in the RulesetBuilder.

Reply:
So there should only be a warning, if two rules are referenced via the exact same `ref` instead of being contained in a ruleset? 
So this would all produce a warning each, but your example from above shouldn't?
```xml
<rule ref="myruleset.xml" />
<rule ref="myruleset.xml" />

<rule ref="myruleset.xml/SomeRule" />
<rule ref="myruleset.xml/SomeRule" />
```

Reply:
I think, that's right: If the override is more specific, then it's ok (as in `myruleset.xml` <-> `myruleset.xml/SomeRule`).

We probably should add some unit tests for these scenarios to pin down, in which case we expect which warning messages.